From 5e38ead2b13d0cf9477aacee2d75432fe0f2176d Mon Sep 17 00:00:00 2001
Message-Id: <5e38ead2b13d0cf9477aacee2d75432fe0f2176d.1389014116.git.minovotn@redhat.com>
In-Reply-To: <c8cc35838d42aa286242772d97e3a9be7bb786ba.1389014116.git.minovotn@redhat.com>
References: <c8cc35838d42aa286242772d97e3a9be7bb786ba.1389014116.git.minovotn@redhat.com>
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Mon, 9 Dec 2013 14:09:35 +0100
Subject: [PATCH 47/50] block/iscsi: set bs->bl.opt_transfer_length

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1386598178-11845-50-git-send-email-pbonzini@redhat.com>
Patchwork-id: 56086
O-Subject: [RHEL 7.0 qemu-kvm PATCH 49/52] block/iscsi: set bs->bl.opt_transfer_length
Bugzilla: 1039557
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Fam Zheng <famz@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>

From: Peter Lieven <pl@kamp.de>

Reviewed-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Peter Lieven <pl@kamp.de>
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
(cherry picked from commit 7572ddc8db114d8c437a97ca3eaedab397f66cda)
---
 block/iscsi.c | 3 +++
 1 file changed, 3 insertions(+)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 block/iscsi.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/block/iscsi.c b/block/iscsi.c
index 94c5af8..fce909b 100644
--- a/block/iscsi.c
+++ b/block/iscsi.c
@@ -1459,6 +1459,9 @@ static int iscsi_open(BlockDriverState *bs, QDict *options, int flags,
         }
         bs->bl.write_zeroes_alignment = sector_lun2qemu(iscsilun->bl.opt_unmap_gran,
                                                         iscsilun);
+
+        bs->bl.opt_transfer_length = sector_lun2qemu(iscsilun->bl.opt_xfer_len,
+                                                     iscsilun);
     }
 
 #if defined(LIBISCSI_FEATURE_NOP_COUNTER)
-- 
1.7.11.7

